src/lib/contact/contactClasses/programContact.ts

```diff
@@ -129,6 +134,35 @@ class Computing {
 			return { success: false, message: 'Failed to upload program' };
 		}
 	}
+
+	public async addToContact(contactAddress: string, mainProgram: IPCProgram): Promise<ResponseType> {
+		try {
+			const index = this.contact.contacts.findIndex((contact) => contact.address === contactAddress);
```

Comment:
you can use the `.find` method that will directly return the element instead of the index